Framework/Algorithms/src/SampleCorrections/SparseWorkspace.cpp

```diff
@@ -132,7 +132,12 @@ SparseWorkspace::SparseWorkspace(const API::MatrixWorkspace &modelWS, const size
   if (eMode == Kernel::DeltaEMode::Direct) {
     mutableRun().addProperty("Ei", modelWS.getEFixed());
   } else if (eMode == Kernel::DeltaEMode::Indirect) {
-    const auto &detIDs = modelWS.detectorInfo().detectorIDs();
+    std::vector<detid_t> detIDs;
+    for (size_t i = 0; i < modelWS.getNumberHistograms(); i++) {
+      const auto &spec = modelWS.getSpectrum(i);
```

Comment:
Do we need a check for `isMonitor` or `hasDetector` before trying to retrieve the detector ID?

Reply:
If the spectrum doesn't have any detectors then the getDetectorIDs calls will return nothing and the code will just insert nothing - so that part is fine I think. I'll have a think about the monitor case. I'm pretty sure the IRIS workspace that Spencer used to find this problem has a monitor so I'll see what that's actually doing in this bit of code

Reply:
The IRIS workspace that Spencer supplied has been partially reduced in Mantid via the Indirect Reduction UI. The monitor spectrum seems to have been removed by the reduction so it's not showing up in this piece of code. Had it not been removed though I think it would have caused a problem because I don't think the monitor has an efixed value - the efixed is written to a comp assembly object ("analyzer") in the instrument definition that's a parent of all the real detectors but a parent of the monitor detector. So to be safe and more general I will add an exclusion on monitors.

Reply:
@DannyHindson @RichardWaiteSTFC IRIS & OSIRIS reduced data files which are used here do not indeed have the monitor spectrum.